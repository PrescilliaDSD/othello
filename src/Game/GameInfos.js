import React from 'react'
import PropTypes from 'prop-types'

const GameInfos = ({ currentPlayer }) => {
  return (
    <aside className="aside">
      <div className={`aside__chip ${currentPlayer}-chip chip`}></div>
      <button type="button" className="pass-button">passer</button>
    </aside>
  )
}

GameInfos.propTypes = {
  currentPlayer: PropTypes.string.isRequired
}

export default GameInfos
