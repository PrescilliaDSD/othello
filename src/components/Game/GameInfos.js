import React from 'react'
import PropTypes from 'prop-types'

const GameInfos = ({ currentPlayer, setCurrentPlayer, opponent, setOpponent, board, lastPlayerPass, setLastPlayerPass, whiteScore, blackScore, endGame, endMsg, player1, player2 }) => {
  const userShouldPass = () => {
    if (board.every((row) => {
      return row.every((column) => {
        return column !== 'available'
      })
    })) {
      if (lastPlayerPass === opponent || lastPlayerPass === currentPlayer) {
        endGame()
      } else {
        setLastPlayerPass(currentPlayer)
        setCurrentPlayer(opponent)
        setOpponent(currentPlayer)
      }
    }
  }

  return (
    <aside className="aside">
      <div className="msg">{endMsg}</div>
      <p>Score : </p>
      <ul className="score__list">
        <li className="score__list__item">{player1} <div className='aside__chip black-chip chip'><p>{blackScore}</p></div></li>
        <li className="score__list__item">{player2} <div className='aside__chip white-chip chip'><p>{whiteScore}</p></div></li>
      </ul>
      <div className="turn">C&apos;est au tour de : <div className={`aside__chip ${currentPlayer}-chip chip`}></div></div>
      <button onClick={userShouldPass} type="button" className="pass-button">passer</button>
    </aside>
  )
}

GameInfos.propTypes = {
  // string
  currentPlayer: PropTypes.string.isRequired,
  opponent: PropTypes.string.isRequired,
  endMsg: PropTypes.string.isRequired,
  player1: PropTypes.string.isRequired,
  player2: PropTypes.string.isRequired,
  lastPlayerPass: PropTypes.string.isRequired,

  // func
  setCurrentPlayer: PropTypes.func.isRequired,
  setOpponent: PropTypes.func.isRequired,
  endGame: PropTypes.func.isRequired,
  setLastPlayerPass: PropTypes.func.isRequired,

  // array
  board: PropTypes.array.isRequired,

  // number
  whiteScore: PropTypes.number.isRequired,
  blackScore: PropTypes.number.isRequired
}

export default GameInfos
