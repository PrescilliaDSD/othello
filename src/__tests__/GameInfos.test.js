import React from 'react'
import renderer from 'react-test-renderer'

import GameInfos from '../components/Game/GameInfos'

describe('GameInfos component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<GameInfos currentPlayer='black' setCurrentPlayer={() => {}} opponent='white' setOpponent={() => {}} board={[]} lastPlayerPass='none' setLastPlayerPass={() => {}} whiteScore={2} blackScore={2} endGame={() => {}} endMsg='' player1='' player2='' />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render an aside tag', () => {
    const tree = renderer
      .create(<GameInfos currentPlayer='black' setCurrentPlayer={() => {}} opponent='white' setOpponent={() => {}} board={[]} lastPlayerPass='none' setLastPlayerPass={() => {}} whiteScore={2} blackScore={2} endGame={() => {}} endMsg='' player1='' player2='' />)
      .toJSON()
    expect(tree.type).toBe('aside')
  })

  it('should render a div in the aside tag', () => {
    const tree = renderer
      .create(<GameInfos currentPlayer='black' setCurrentPlayer={() => {}} opponent='white' setOpponent={() => {}} board={[]} lastPlayerPass='none' setLastPlayerPass={() => {}} whiteScore={2} blackScore={2} endGame={() => {}} endMsg='' player1='' player2='' />)
      .toJSON()
    expect(tree.children[0].type).toBe('div')
  })

  it('should render a black chip in the div tag', () => {
    const tree = renderer
      .create(<GameInfos currentPlayer='black' setCurrentPlayer={() => {}} opponent='white' setOpponent={() => {}} board={[]} lastPlayerPass='none' setLastPlayerPass={() => {}} whiteScore={2} blackScore={2} endGame={() => {}} endMsg='' player1='' player2='' />)
      .toJSON()
    expect(tree.children[3].children[1].props.className).toBe('aside__chip black-chip chip')
  })

  it('should render a button in the aside tag', () => {
    const tree = renderer
      .create(<GameInfos currentPlayer='black' setCurrentPlayer={() => {}} opponent='white' setOpponent={() => {}} board={[]} lastPlayerPass='none' setLastPlayerPass={() => {}} whiteScore={2} blackScore={2} endGame={() => {}} endMsg='' player1='' player2='' />)
      .toJSON()
    expect(tree.children[4].type).toBe('button')
  })

  it('should render a button in the aside tag with content "passer"', () => {
    const tree = renderer
      .create(<GameInfos currentPlayer='black' setCurrentPlayer={() => {}} opponent='white' setOpponent={() => {}} board={[]} lastPlayerPass='none' setLastPlayerPass={() => {}} whiteScore={2} blackScore={2} endGame={() => {}} endMsg='' player1='' player2='' />)
      .toJSON()
    expect(tree.children[4].children[0]).toBe('passer')
  })

  it('should render a black chip if player 1 is active', () => {
    const tree = renderer
      .create(<GameInfos currentPlayer='black' setCurrentPlayer={() => {}} opponent='white' setOpponent={() => {}} board={[]} lastPlayerPass='none' setLastPlayerPass={() => {}} whiteScore={2} blackScore={2} endGame={() => {}} endMsg='' player1='' player2='' />)
      .toJSON()
    expect(tree.children[3].children[1].props.className).toBe('aside__chip black-chip chip')
  })

  it('should render a white chip if player 2 is active', () => {
    const tree = renderer
      .create(<GameInfos currentPlayer='white' setCurrentPlayer={() => {}} opponent='black' setOpponent={() => {}} board={[]} lastPlayerPass='none' setLastPlayerPass={() => {}} whiteScore={2} blackScore={2} endGame={() => {}} endMsg='' player1='' player2='' />)
      .toJSON()
    expect(tree.children[3].children[1].props.className).toBe('aside__chip white-chip chip')
  })
})
