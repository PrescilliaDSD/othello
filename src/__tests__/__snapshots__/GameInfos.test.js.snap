// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameInfos component should render correctly 1`] = `
<aside
  className="aside"
>
  <div
    className="msg"
  >
    
  </div>
  <p>
    Score : 
  </p>
  <ul
    className="score__list"
  >
    <li
      className="score__list__item"
    >
      
       
      <div
        className="aside__chip black-chip chip"
      >
        <p>
          2
        </p>
      </div>
    </li>
    <li
      className="score__list__item"
    >
      
       
      <div
        className="aside__chip white-chip chip"
      >
        <p>
          2
        </p>
      </div>
    </li>
  </ul>
  <div
    className="turn"
  >
    C'est au tour de : 
    <div
      className="aside__chip black-chip chip"
    />
  </div>
  <button
    className="pass-button"
    onClick={[Function]}
    type="button"
  >
    passer
  </button>
</aside>
`;
